app/models/incomeSourceDetails/viewmodels/ObligationsViewModel.scala

```diff
@@ -16,10 +16,116 @@
 
 package models.incomeSourceDetails.viewmodels
 
+import models.incomeSourceDetails.TaxYear
+
 import java.time.LocalDate
 
 final case class ObligationsViewModel(quarterlyObligationsDates: Seq[Seq[DatesModel]],
-                                      finalDeclarationDates: Seq[DatesModel], currentTaxYear: Int, showPrevTaxYears: Boolean)
+  finalDeclarationDates: Seq[DatesModel], currentTaxYear: Int, showPrevTaxYears: Boolean
+) {
+
+  def getOverdueObligationsMessageComponents(currentDate: LocalDate, isBusinessHistoric: Boolean): OverdueObligationsMessageComponents = {
+    if (isBusinessHistoric) {
+      val nonHistoricOverdueObligations = getNumberOfNonHistoricOverdueObligations(currentDate)
+      if (nonHistoricOverdueObligations > 1) {
+        OverdueObligationsMessageComponents("obligation.inset.multiple-historic-overdue.text",
+          List(getNumberOfNonHistoricOverdueObligations(currentDate).toString, (currentTaxYear - 2).toString, (currentTaxYear - 1).toString))
```

Comment:
What do to years look like from here, 24 & then 22/23?

Reply:
Yes that's right `currentTaxYear` in the view model is the end year of the current tax year. So in the case of the current tax year being 2023-2024 with a historic business, the message should say that payments need to be made before the previous tax year of 2022-2023.